blockchain/utxobackend.go

```diff
@@ -22,6 +28,9 @@ type UtxoBackend interface {
 	// both the entry and the error.
 	FetchEntry(outpoint wire.OutPoint) (*UtxoEntry, error)
 
+	// FetchState returns the current state of the UTXO set.
+	FetchState() (*utxoSetState, error)
```

Comment:
Similar to `PutUtxos`, since this is an interface, the returned type should be exported as it would otherwise be unusable outside of the package.

Also, I'm kind of torn on the naming of this one.  It really is more like `FetchTip`, but I guess a more generic name like `FetchState` will allow other things to be added to it if necessary without needing additional rototilling.

Reply:
Good catch, I updated the `UtxoSetState` and `UtxoBackendInfo` types to be exported since they are part of the interface definition.  I also noticed that the `viewFilteredSet` was not exported and is used in the `UtxoCacher` interface, so I added a commit to export that as well. 

I also hear you on the naming.  I'm open to changing it but had it more generic for exactly the reason you mention, that we may want to add other things to it (such as whether a flush is in progress, which would be needed if we want to move to more granular flushes than flushing the whole cache in a single transaction as it currently does).

Reply:
Sounds good.  I wasn't really convinced one way or the other, but especially since you already have things in mind you can envision adding to it, going with state seems to be the prudent choice.